source/features/useful-not-found-page.js

```diff
@@ -0,0 +1,105 @@
+/*
+This feature adds more useful 404 (not found) page.
+- Display the full URL clickable piece by piece
+- Strikethrough all anchor that return a 404 status code
+*/
+
+import {h} from 'dom-chef';
+import select from 'select-dom';
+import {getCleanPathname} from '../libs/page-detect';
+import getDefaultBranch from '../libs/get-default-branch';
+
+async function is404(url) {
+	const {status} = await fetch(url, {method: 'head'});
+	return status === 404;
+}
+
+function getStrikeThrough(text) {
+	return <del style={{color: '#6a737d'}}>{text}</del>;
+}
+
+async function checkAnchor(anchor) {
+	if (await is404(anchor.href)) {
+		anchor.replaceWith(getStrikeThrough(anchor.textContent));
+	}
+}
+
+function parseCurrentURL() {
+	const parts = getCleanPathname().split('/');
+	if (parts[2] === 'blob') { // Blob URLs are never useful
+		parts[2] = 'tree';
+	}
+	return parts;
+}
+
+// If the resource was deleted, link to the commit history
+async function addCommitHistoryLink(bar) {
+	const parts = parseCurrentURL();
+	if (parts[2] !== 'tree') {
+		return;
+	}
+	parts[2] = 'commits';
+	const url = '/' + parts.join('/');
+	if (await is404(url)) {
+		return;
+	}
+	bar.after(
+		<p class="container">
+			See also the file’s {<a href={url}>commit history</a>}
+		</p>
+	);
+}
+
+// If the resource exists in the default branch, link it
+async function addDefaultBranchLink(bar) {
+	const parts = getCleanPathname().split('/');
+	const [,,, branch] = parts;
+	if (!branch) {
+		return;
+	}
+	const defaultBranch = await getDefaultBranch();
+	if (!defaultBranch || branch === defaultBranch) {
+		return;
+	}
+	parts[3] = defaultBranch; // Change branch
+	const url = '/' + parts.join('/');
+	if (await is404(url)) {
+		return;
+	}
+	bar.after(
+		<p class="container">
+			See also the file on the {<a href={url}>default branch</a>}
+		</p>
+	);
+}
+
+export default function () {
+	const parts = parseCurrentURL();
+	const bar = <h2 class="container" />;
+
+	for (let i = 0; i < parts.length; i++) {
```

Comment:
```js
for (const [i, part] of parts.entries()) {
```

Reply:
Thought so, but just a few lines later I’m on a similar loop that can’t easily use for-of (it comes out longer) so I left this one for consistency. What do you think?

Reply:
Readability > Consistency, and consistency between loop statements doesn't matter much. I still think this should be changed.